### ISSUE TITLE
--get-server-output on client doesn't recieve last few fields from server

### ISSUE THREAD

**Issue by nomel** [esnet/iperf#251]
When --get-server-output is used, the last interval and server end statistics are missing, with --json or without.

Using --json:
The end field is always blank, as seen on client (it's populated correctly on the server):
```
                "end":  {
                }
```

And the last interval from the server is missing (it's populated on the server). As an example, using `-i0` on the server, no intervals are reported since the last (and only), interval is missing:

```
                "intervals":    [],
                "end":  {
                }
```

Things are the same without --json. With a few server side intervals, server says:
```
-----------------------------------------------------------
Server listening on 5201
-----------------------------------------------------------
Accepted connection from 10.10.11.130, port 57715
[  5] local 10.10.11.143 port 5201 connected to 10.10.11.130 port 54493
[ ID] Interval           Transfer     Bandwidth       Jitter    Lost/Total Datagrams
[  5]   0.00-5.00   sec   304 KBytes   498 Kbits/sec  1.409 ms  0/38 (0%)
[  5]   5.00-10.00  sec   304 KBytes   498 Kbits/sec  0.175 ms  0/38 (0%)
[  5]  10.00-13.04  sec   184 KBytes   497 Kbits/sec  0.082 ms  0/23 (0%)
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Jitter    Lost/Total Datagrams
[  5]   0.00-13.04  sec   800 KBytes   503 Kbits/sec  0.082 ms  0/99 (0%)
```

But client sees, with both missing last interval and summary:
```
Server output:
Accepted connection from 10.10.11.130, port 57715
[  5] local 10.10.11.143 port 5201 connected to 10.10.11.130 port 54493
[ ID] Interval           Transfer     Bandwidth       Jitter    Lost/Total Datagrams
[  5]   0.00-5.00   sec   304 KBytes   498 Kbits/sec  1.409 ms  0/38 (0%)
[  5]   5.00-10.00  sec   304 KBytes   498 Kbits/sec  0.175 ms  0/38 (0%)
```

And, with `-i0`, just as with --json, last and only interval are missing, and no summary. Server sees:
```
-----------------------------------------------------------
Server listening on 5201
-----------------------------------------------------------
Accepted connection from 10.10.11.130, port 57714
[  5] local 10.10.11.143 port 5201 connected to 10.10.11.130 port 56001
[ ID] Interval           Transfer     Bandwidth       Jitter    Lost/Total Datagrams
[  5]   0.00-13.04  sec   792 KBytes   498 Kbits/sec  0.074 ms  0/99 (0%)
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Jitter    Lost/Total Datagrams
[  5]   0.00-13.04  sec   800 KBytes   503 Kbits/sec  0.074 ms  0/99 (0%)
```

Client sees:
```
Server output:
-----------------------------------------------------------
Server listening on 5201
-----------------------------------------------------------
Accepted connection from 10.10.11.130, port 57714
[  5] local 10.10.11.143 port 5201 connected to 10.10.11.130 port 56001
```

**Comment by nomel** [comment]
From an teensy glance at the code, looks like the summary and interval is constructed in `iperf_reporter_callback()` when the test state is `DISPLAY_RESULTS`, but this happens after `iperf_exchange_results()` at the end of the test (test state of `TEST_END` in `iperf_handle_message_server()`). I think the last interval is missing since the `server_reporter_timer_proc()` timer didn't have a chance to fire. So, a `reporter_callback()`  probably just has to be added before the exchange call to tidy up the results (with a `TEST_END` flowthrough case for `DISPLAY_RESULTS` in` iperf_reporter_callback()`).

I could be wrong about all of this, but hope it helps.

**Comment by bmah888** [comment]
Hrm, yep this is a real bug.

I haven't dug into your analysis yet but it sounds reasonable.

**Comment by bmah888** [comment]
Fixed on master and merged to the 3.0-STABLE branch.  Thanks!
